mono/metadata/threads.c

```diff
@@ -5993,7 +5993,7 @@ mono_threads_summarize_one (MonoThreadSummary *out, MonoContext *ctx)
 
 	MonoInternalThread *thread = (MonoInternalThread *) mono_gchandle_get_target (handle);
 
```

Comment:
Why? The previous form is resilient to changing type, assuming the entire thing should be zeroed.

Reply:
I can revert this change if people really prefer the previous one, but as a reader I find the `*out` form produces code that is much less readable. For short functions like this, you can just visually check out the variable type, but reminders of types become more and more useful to code reviewers as the code gets longer.

I also prefer it because the old code is really not resilient to a changing type signature. If you make `out` into a `gpointer *`, this program will subtilely break. 

Now one can say that a programmer who makes that change will of course update the memset. I favor defensive programming in general though. C makes it easy enough to shoot yourself in the foot without me adding more footguns.

Reply:
The less type-dependent form is generally considered more resilient and readable and defensive, but there are subtle arguments either way.

The problem with naming the type, is the reader has to double check if it is in fact the correct type. And the writer has to notice to update it.

Changing out to gpointer* works, depending on intent. You'd be nulling the pointer.
Try to use strong types instead of gpointer, as well.